cli/src/pcluster/validators/slurm_settings_validator.py

```diff
@@ -10,40 +10,86 @@
 # limitations under the License.
 
 from enum import Enum
+from typing import Dict, List
 
 from pcluster.validators.common import FailureLevel, Validator
 
 # SLURM SETTINGS are case-insensitive - keep them lowercase since they are compared with setting.lower()
 SLURM_SETTINGS_DENY_LIST = {
-    "Queue": ["nodes", "partitionname", "resumetimeout", "state", "suspendtime"],
-    "ComputeResource": ["cpus", "features", "gres", "nodeaddr", "nodehostname", "nodename", "state"],
+    "SlurmConf": {
+        "Global": [
+            "slurmctldhost",
+            "suspendtime",
+            "resumetimeout",
+            "slurmuser",
+            "reconfigflags",
+            "launchparameters",
+            "slurmctldparameters",
+            "treewidth",
+            "communicationparameters",
+            "suspendprogram",
+            "resumeprogram",
+            "resumefailprogram",
+            "grestypes",
+            "slurmctldlogfile",
+            "slurmdlogfile",
+            "taskplugin",
+            "jobcomppass",
+        ],
+        "Accounting:": [
+            "accountingstoragetype",
+            "accountingstoragehost",
+            "accountingstorageport",
+            "accountingstorageuser",
+            "jobacctgathertype",
+        ],
+    },
+    "Queue": {
+        "Global": ["nodes", "partitionname", "resumetimeout", "state", "suspendtime"],
+    },
+    "ComputeResource": {
+        "Global": ["cpus", "features", "gres", "nodeaddr", "nodehostname", "nodename", "state"],
+    },
 }
 
 
-class SlurmCustomSettingLevel(str, Enum):
+class CustomSlurmSettingLevel(str, Enum):
     """
-    Slurm Custom Settings level.
+    Custom Slurm Settings level.
 
     This enum defines the scope where the custom settings are defined.
     """
 
+    SLURM_CONF = "SlurmConf"
```

Comment:
Q: having here a more fine grained list o levels, like `SlurmConf.Global` and `SlurmConf.Accounting` would make the messaging more easy to understand or it would simply add some noise ?

Reply:
The Context entity I added is more of an internal representation of when parameters are used more than something immediately recognizable by the end users (such as the Level you had already introduced).

I would say it would not be too useful to add it to the message.

Unless we want to make it explicit that some parameters are not allowed when our Slurm accounting integration is enabled. But for that we would need to re-write the messaging (possibly using a separate validator function).

Reply:
Let's think about this.

Reply:
Ok. I'm fine with this.
We will change this behavior if/when there will be a clear need for it.